src/HttpServer.ts

```diff
@@ -65,31 +65,30 @@ export class HttpServer {
     }
 
     // TODO: use this when changing to jsonrpc
-    async rootHandler(req: any, res: any): Promise<void> {
+    async rootHandler(req: Request, res: Response): Promise<void> {
         let status;
+        let id = -1;
         try {
-            let res;
-            // @ts-ignore
-            const func = this.backend[req.body.method];
-            if (func != null) {
-                res = (await func.apply(this.backend, [req.body.params])) ?? {
-                    code: 200
-                };
-            } else {
-                // @ts-ignore
-                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-                throw Error(
-                    `Implementation of method ${req.body.params} not found on backend!`
-                );
+            if (
+                !Object.prototype.hasOwnProperty.call(req.body, 'id') ||
+                !Object.prototype.hasOwnProperty.call(req.body, 'method')
```

Comment:
I really don't see the need for this.
1. From tests I see that the only thing you expect to fail on is when the request has no `body` so: `if (!requestBody)` should suffice.
2. If you want to check for properties, why not just use something like `if (!(body?.id && body.method && body.params))`?
3. Since the try-catch block has been here before, I add this comment here: don't. Write middleware for specialised error handling

Reply:
updated